firebase-appdistribution-api/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -0,0 +1,278 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appdistribution;
+
+import android.app.Activity;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.gms.tasks.OnCanceledListener;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.appdistribution.FirebaseAppDistributionException.Status;
+import com.google.firebase.appdistribution.internal.FirebaseAppDistributionService;
+import com.google.firebase.inject.Provider;
+import java.util.concurrent.Executor;
+
+/**
+ * The Firebase App Distribution API provides methods to update the app to the most recent
+ * pre-release build.
+ *
+ * <p>By default, Firebase App Distribution is automatically initialized.
+ *
+ * <p>Call {@link FirebaseAppDistribution#getInstance()} to get the singleton instance of
+ * FirebaseAppDistribution.
+ */
+public class FirebaseAppDistribution {
```

Comment:
Could this be an interface, instead of FirebaseAppDistributionService?

Reply:
Done